sympy/matrices/expressions/hadamard.py

```diff
@@ -37,7 +37,6 @@ def hadamard_product(*matrices):
     if len(matrices) == 1:
         return matrices[0]
     else:
-        matrices = [i for i in matrices if not i.is_Identity]
```

Comment:
Weird that this line was ever here in the first place. Maybe someone incorrectly copied it from MatMul?